packages/redux-api-auth-middleware/src/auth.middleware.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import type { Dispatch, Store } from 'redux';
+import { RSAA } from 'redux-api-middleware';
+
+import { refreshTokenAction } from './store/actions';
+import type { Action, GetState, MiddlewareConfig } from './types';
+import { isTokenExpired } from './helpers';
+
+let refreshPromise: Promise<any> | null = null;
+
+export default function({
+  authConfig = { header: 'Authorization', type: 'Bearer' },
+  refreshConfig,
+}: MiddlewareConfig = {}) {
+  if (refreshConfig && !refreshConfig.endpoint) {
+    throw new Error('You must set the endpoint for refreshing token.');
+  }
+
+  return (store: Store<GetState, Action, Dispatch<Action>>) => {
+    if (!store.getState().auth) {
```

Comment:
I think it would be nice to make used reducer's name configurable. It could be a future improvement.

Reply:
+1